src/client/components/dimension-list-tile/dimension-list-tile.tsx

```diff
@@ -43,31 +42,45 @@ export interface DimensionListTileProps extends React.Props<any> {
 export interface DimensionListTileState {
   menuOpenOn?: Element;
   menuDimension?: Dimension;
-  highlightDimension?: Dimension;
   showSearch?: boolean;
   searchText?: string;
 }
 
+const hasSearchTextPredicate = (searchText: string) => (dimension: Dimension): boolean => {
+  return dimension.title.toLowerCase().includes(searchText.toLowerCase());
+};
+
+const isFilteredOrSplitPredicate = (essence: Essence) => (dimension: Dimension): boolean => {
+  const { dataCube, filter, splits } = essence;
+  const filteredDimensions = filter.clauses.map((clause) => dataCube.dimensions.getDimensionByExpression(clause.expression));
+  const splitDimensions = splits.splitCombines.map((split) => dataCube.dimensions.getDimensionByExpression(split.expression));
+
+  return filteredDimensions.contains(dimension) || splitDimensions.contains(dimension);
```

Comment:
skoro tu jest OR, to jeżeli pierwszy warunek jest spełniony, cała praca potrzebna do zrobienia `splitDimensions` idzie na marne. IMO do dwóch osobnych funkcji i później:

```
return isFilteredPredicate(essence) || isSplitPredicate(essence);
```

Reply:
👍